Dockerfile

```diff
@@ -1,19 +1,14 @@
-# This is designed to be run from fig as part of a
-# Marketplace development environment.
-
 # NOTE: this is not provided for production usage.
 
-FROM  mozillamarketplace/centos-python27-mkt:0.6
-
-RUN yum install -y supervisor
+FROM  mozillamarketplace/centos-python27-mkt:latest
 
-RUN mkdir -p /pip/{cache,build}
+RUN yum install -y supervisor && yum clean all
 
-ADD requirements /pip/requirements
+COPY requirements /srv/webpay/requirements
+RUN pip install --no-deps -r /srv/webpay/requirements/docker.txt --find-links https://pyrepo.addons.mozilla.org/
 
-# Setting cwd to /pip ensures egg-links for git installed deps are created in /pip/src
-WORKDIR /pip
-RUN pip install -b /pip/build --download-cache /pip/cache --no-deps -r /pip/requirements/docker.txt --find-links https://pyrepo.addons.mozilla.org/
+COPY . /srv/webpay
```

Comment:
If you do then each change to the source code Python or whatever changes the layer and then the pip install is triggered. This way the pip install is only triggered when requirements.txt changes, which is a lot less common.